### ISSUE TITLE
Hovering a prefetch Link causes unnecessary re-render of the error boundary in production

### ISSUE THREAD

**Issue by ChaserZ98** [vercel/next.js#63159]
### Link to the code that reproduces this issue

https://github.com/ChaserZ98/next-error-link-rerender-bug

### To Reproduce

1. Run `npm run build`
2. Run `npm run start`
3. Visit any page which may produce an error such as http://localhost:3000/link1 of the reproduction project.
4. Trigger the error such as clicking the button in http://localhost:3000/link1 page of the reproduction project.
5. Hover any Link with prefetch enabled.

## Current behavior
- Hovering any Link with prefetch enabled will trigger a re-render of the error boundary but non-prefetch Links won't trigger such re-render.
- As for the reproduction project, I include `console.log("error render")` in the error boundary so that the console will print `error render` each time the error boundary is rendered. So in this case, the users will see the console printing `error render` when they hovering any prefetch Link.
- Such behavior is present in the production build but not present in the development stage, i.e. `npm run dev`.

## Expected behavior
Hovering any Link that does not trigger a state change should not cause the error boundary to re-render, no matter it is a prefetch link or non-prefetch Link.

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Home
Binaries:
  Node: v18.18.2
  npm: 10.2.4
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.2.0-canary.13
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router

### Which stage(s) are affected? (Select all that apply)

next build (local), next start (local)

### Additional context

The behavior is present in release 14.0.2-canary.7 and above.
A similar issue might be #58788 . But the fix for that issue isn't working for this one since such behavior still presents in the latest canary release, which is 14.2.0-canary.15 by the time this issue is posted.

**Comment by samcx** [comment]
@ChaserZ98 Can confirm this is an issue. We are taking a look!

**Comment by ChaserZ98** [comment]
@samcx Thanks! Let me know if any details are needed for this issue. Looking forward to a fix!

**Comment by samcx** [comment]
@ChaserZ98 This should be <picture data-single-emoji=":fixed:" title=":fixed:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/fixed/66bb8455f24bd50e.png" alt=":fixed:" align="absmiddle"></picture> in [v14.3.0-canary.7](https://github.com/vercel/next.js/releases/tag/v14.3.0-canary.7)! Let us know if you can confirm.

**Comment by ChaserZ98** [comment]
@samcx I did the check in `v14.3.0-canary.7` and the latest canary version, which is `v14.3.0-canary.8` by the time of this reply. Such re-render behavior caused by prefetch Link does not show up anymore. The issue is solved. Perfect work! Thanks!
